apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.command.ts

```diff
@@ -0,0 +1,9 @@
+import { EnvironmentWithUserObjectCommand } from '@novu/application-generic';
+import { CreateWorkflowDto } from '../../dto/workflow.dto';
+
+export type RequiredProp<T, K extends keyof T> = T & { [P in K]-?: T[P] };
+
+export class UpsertWorkflowCommand extends EnvironmentWithUserObjectCommand {
+  _id?: string;
+  workflowDto: RequiredProp<CreateWorkflowDto, 'origin'>;
```

Comment:
I wonder if we should remove all the `workflowDto` objects from all the new commands and make the command object flatter. 




pseudo in order to illustrate the comment above:
```
export class UpsertWorkflowCommand extends EnvironmentWithUserObjectCommand {
  _id?: string;
  {...workflowDto}
  }
  ```
  
  
  
 what do you think?

Reply:
I don't see value in flattening it, also i'm not totally sure this syntax is allowed, I find that it makes sense this differentiation, I think it makes it more clear that way , can you share your reasoning

Reply:
resolved in sync conversation